src/Bicep.Core.IntegrationTests/Decorators/DecoratorTests.cs

```diff
@@ -340,5 +340,41 @@ public void UnfinishedDecoratorInResourceBody_ShouldPromptForNamespaceOrDecorato
                 template.Should().BeNull();
             }
         }
+
+        [TestMethod]
+        public void DescriptionDecorator_OnMetadata_ShouldProduceError()
+        {
+            var (template, diagnostics, _) = CompilationHelper.Compile(@"
+@description('description')
+metadata foo = 'foo'
+");
+            using (new AssertionScope())
+            {
+                diagnostics.ExcludingLinterDiagnostics().Should().HaveDiagnostics(new[]
+                {
+                    ("BCP269", DiagnosticLevel.Error, "Function \"description\" cannot be used as a metadata decorator."),
+                });
+                template.Should().BeNull();
+            }
```

Comment:
There's a more concise way to write this - can you try and follow other patterns in this repo?